src/gui/qgsdatumtransformdialog.cpp

```diff
@@ -190,9 +248,123 @@ QgsDatumTransformDialog::~QgsDatumTransformDialog()
+  bool foundNonDeprecated = false;
+  QPair<QPair<QgsCoordinateReferenceSystem, int>, QPair<QgsCoordinateReferenceSystem, int> > fallback;
+  fallback.first.first = mSourceCrs;
+  fallback.second.first = mDestinationCrs;
+  bool foundFallback = false;
+
+  for ( const QgsDatumTransform::TransformPair &transform : qgis::as_const( mDatumTransforms ) )
+  {
+    if ( transform.sourceTransformId == -1 && transform.destinationTransformId == -1 )
+      continue;
+
+    const QgsDatumTransform::TransformInfo srcInfo = QgsDatumTransform::datumTransformInfo( transform.sourceTransformId );
+    const QgsDatumTransform::TransformInfo destInfo = QgsDatumTransform::datumTransformInfo( transform.destinationTransformId );
+    if ( !foundPreferredNonDeprecated && ( ( srcInfo.preferred && !srcInfo.deprecated ) || transform.sourceTransformId == -1 )
+         && ( ( destInfo.preferred && !destInfo.deprecated ) || transform.destinationTransformId == -1 ) )
+    {
+      preferredNonDeprecated.first.second = transform.sourceTransformId;
+      preferredNonDeprecated.second.second = transform.destinationTransformId;
+      foundPreferredNonDeprecated = true;
+    }
+    else if ( !foundPreferred && ( srcInfo.preferred || transform.sourceTransformId == -1 ) &&
+              ( destInfo.preferred || transform.destinationTransformId == -1 ) )
+    {
+      preferred.first.second = transform.sourceTransformId;
+      preferred.second.second = transform.destinationTransformId;
+      foundPreferred = true;
+    }
+    else if ( !foundNonDeprecated && ( !srcInfo.deprecated || transform.sourceTransformId == -1 )
+              && ( !destInfo.deprecated || transform.destinationTransformId == -1 ) )
+    {
+      nonDeprecated.first.second = transform.sourceTransformId;
+      nonDeprecated.second.second = transform.destinationTransformId;
+      foundNonDeprecated = true;
+    }
+    else if ( !foundFallback )
+    {
+      fallback.first.second = transform.sourceTransformId;
+      fallback.second.second = transform.destinationTransformId;
+      foundFallback = true;
+    }
+  }
+  if ( foundPreferredNonDeprecated )
+    return preferredNonDeprecated;
+  else if ( foundPreferred )
+    return preferred;
+  else if ( foundNonDeprecated )
+    return nonDeprecated;
+  else
+    return fallback;
+}
+
+void QgsDatumTransformDialog::applyDefaultTransform()
+{
+  if ( mDatumTransforms.count() > 0 )
+  {
+    QgsCoordinateTransformContext context = QgsProject::instance()->transformContext();
+    const QPair<QPair<QgsCoordinateReferenceSystem, int>, QPair<QgsCoordinateReferenceSystem, int> > dt = defaultDatumTransform();
+    context.addSourceDestinationDatumTransform( dt.first.first, dt.second.first, dt.first.second, dt.second.second );
+    QgsProject::instance()->setTransformContext( context );
+  }
 }
 
 QPair<QPair<QgsCoordinateReferenceSystem, int>, QPair<QgsCoordinateReferenceSystem, int> > QgsDatumTransformDialog::selectedDatumTransforms()
```

Comment:
A bit off-topic, but it would be nice to have this return a struct instead of QPairs.